packages/gitbook-v2/src/app/~gitbook/revalidate/route.ts

```diff
@@ -0,0 +1,42 @@
+import { type NextRequest, NextResponse } from 'next/server';
+
+import { GITBOOK_APP_SECRET } from '@v2/lib/env';
+import { revalidateTag } from 'next/cache';
+
+interface JsonBody {
+    tags: string[];
+}
+
+/**
+ * Revalidate cached data based on tags.
+ * The body should be a JSON with { tags: string[] }
+ */
+export async function POST(req: NextRequest) {
+    const json = (await req.json()) as JsonBody;
+
+    if (GITBOOK_APP_SECRET && req.headers.get('x-gitbook-secret') !== GITBOOK_APP_SECRET) {
```

Comment:
This is a weak security. The good security would be to sign the body with the secret.